lib/engine/game/g_1858/graph.rb

```diff
@@ -0,0 +1,141 @@
+# frozen_string_literal: true
+
+require_relative '../../graph'
+
+module Engine
+  module Game
+    module G1858
+      class Graph < Engine::Graph
+        # TODO: this method has been cut'n'pasted from Engine::Graph, with a
```

Comment:
can we do this refactor?

Reply:
I've added two commits to do this.

- 2fceee9 extracts two methods from `Engine::Graph.compute`. These are used when `@home_is_token` is true and calculate the home hexes and the home nodes that should be used to start the graphs.
- 26fcea0 overrides `home_hex_nodes` in `G1858::Graph`. This is where the implementation is different from the existing code. Two things are changed, how multiple cities in a hex are handled, and also being able to start a node walk from a section of plain track.